util/reggen/gen_tock.py

```diff
@@ -0,0 +1,384 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Generate Rust constants from validated register JSON tree
+"""
+
+import io
+import logging as log
+import sys
+import textwrap
+import warnings
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Set, TextIO
+
+from .ip_block import IpBlock
+from .multi_register import MultiRegister
+from .params import LocalParam
+from .register import Register
+from .window import Window
+
+REG_VISIBILITY = 'pub(crate)'
+FIELD_VISIBILITY = 'pub(crate)'
+
+
+def genout(outfile: TextIO, msg: str, *args: Any, **kwargs: Any) -> None:
+    outfile.write(msg.format(*args, **kwargs))
+
+
+def indent(s: str, amount: int = 4) -> str:
+    result = []
+    indent = 0
+    for line in s.splitlines():
+        indented = []
+        comment = False
+        for i, ch in enumerate(line):
+            if i > 0 and ch == '/' and line[i - 1] == '/':
+                comment = True
+            if not comment and ch in ")]}":
+                indent -= 1
+            if i == 0:
+                indented.append(' ' * indent * amount)
+            if not comment and ch in "([{":
+                indent += 1
+            indented.append(ch)
+        result.append(''.join(indented))
+        if line == '':
+            result.append('')
+    return '\n'.join(result)
+
+
+def to_snake_case(s: str) -> str:
+    val = []
+    for i, ch in enumerate(s):
+        if i > 0 and ch.isupper():
+            val.append('_')
+        val.append(ch)
+    return ''.join(val)
```

Comment:
This seems to be a duplicate of [these lines](https://github.com/lowRISC/opentitan/blob/ec30c9fbee87177654c3c5d0999c119875849d73/util/reggen/gen_rust.py#L29-L35). Can you not just import the `gen_rust` module that is in the same package and reuse the `to_snake_case` function defined there?

Reply:
It is a duplicate; These conversion functions more properly belong in their own library.

I intend to find out if anyone is using `gen_rust` and then delete it if not.  Its constants, IMHO, are not particularly useful.

Reply:
SG, thats what I figured after reading through (see below comment) hehe.